website/src/components/showcase/ShowcaseFilterToggle/index.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState, useEffect, useCallback, useRef} from 'react';
+import {useHistory, useLocation} from '@docusaurus/router';
+
+import styles from './styles.module.css';
+
+export type Operator = 'OR' | 'AND';
+
+export const OperatorQueryKey = 'operator';
+
+export function readOperator(search: string): Operator {
+  return (new URLSearchParams(search).get(OperatorQueryKey) ??
+    'OR') as Operator;
+}
+
+export default function ShowcaseFilterToggle(): JSX.Element {
+  const id = `showcase_filter_toggle`;
+  const location = useLocation();
+  const history = useHistory();
+  const [operator, setOperator] = useState(false);
+  const inputRef = useRef<HTMLInputElement>();
+  useEffect(() => {
+    inputRef.current.checked = readOperator(location.search) === 'AND';
```

Comment:
I don't really understand, why use a ref+mutation here instead of toggling state?

Reply:
That was the implementation until @Josh-Cena refactored the components, Maybe there is a reason why he used ref?

Reply:
My React just gets so hacky😅 Now when I look back I see that using state to drive the checked state may be a better choice...